pkg/server/endpoints/registration/handler.go

```diff
@@ -618,6 +622,13 @@ func (h *Handler) prepareRegistrationEntry(entry *common.RegistrationEntry, forU
 	}
 
 	var err error
+	for _, dns := range entry.DnsList {
+		err = validateDNS(dns)
+		if err != nil {
+			return nil, err
```

Comment:
It would be useful to know which DNS name failed validation e.g. `fmt.Errorf("dns name %v failed validation: %v"), dns, err`